src/compiler/parser.ts

```diff
@@ -7168,8 +7169,19 @@ namespace Parser {
                 //
                 // could be legal, it would add complexity for very little gain.
                 case SyntaxKind.InterfaceKeyword:
-                case SyntaxKind.TypeKeyword:
                     return nextTokenIsIdentifierOnSameLine();
+                case SyntaxKind.TypeKeyword:
+                    const isIdentifierOnSameLine = nextTokenIsIdentifierOnSameLine();
+                    if (isIdentifierOnSameLine && token() === SyntaxKind.AsKeyword || token() === SyntaxKind.SatisfiesKeyword) {

```

Comment:
Do you think this part should be in a proper function?
We could return something like this:
```ts
return nextTokenIsIdentifierOnSameLine() && someNameToExplainsWhatIsDoneRegardingSatisfiesAndAs()
```